feat: add sushiswap aggregator adapter

0xMasayoshi opened the pull request:
Rootstock, Boba BNB, Filecoin, HAQQ, Harmony, & SKALE Europa were throwing RPC errors during tests, so they are currently disabled. Any idea why this was happening?

Also dailyVolumes are incorrectly calculated as 0 or very low on some chains, is this due to limited token prices on those chains?

dtmkeng commented:
yes i guess some chain we missing token price,  you can use mapping with gc price if you need

0xMasayoshi commented:
> yes i guess some chain we missing token price, you can use mapping with gc price if you need

I tried using `Balances.addCGToken()`, but token decimals is not included in the response so the price is very inflated. ex: https://coins.llama.fi/prices/current/coingecko:weth

llamatester commented:
The sushiswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts aggregators sushiswap
    
    🦙 Running SUSHISWAP adapter 🦙
    ---------------------------------------------------
    Start Date:	Thu, 04 Jul 2024 13:28:57 GMT
    End Date:	Fri, 05 Jul 2024 13:28:57 GMT
    ---------------------------------------------------
    
    error fetching block [Error: Llama RPC error! methos: undefined 
    - host: https://zetachain-evm.blockpi.network/v1/rpc/public error: Cannot read properties of null (reading 'number')] {
      _underlyingError: '[object Object]',
      _isCustomError: true
    }
    error fetching block [Error: Llama RPC error! methos: undefined 
    - host: https://zetachain-evm.blockpi.network/v1/rpc/public error: Cannot read properties of null (reading 'number')] {
      _underlyingError: '[object Object]',
      _isCustomError: true
    }
    ARBITRUM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 955.47 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    ARBITRUM_NOVA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 17.03 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    AVAX 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 176.59 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    BASE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.39 M
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    BLAST 👇
    Backfill start time: 1/3/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 45.95 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    BOBA 👇
    Backfill start time: 22/3/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 84.72 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    BOBA_BNB 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.72 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    BSC 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 429.68 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    BITTORRENT 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    CELO 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 181.01 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    CORE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 183.37 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    ETHEREUM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 4.49 M
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    FANTOM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.65 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    FUSE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.44 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    XDAI 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.34 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    KAVA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.74 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    LINEA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 206.59 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    METIS 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 95
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    MOONBEAM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 74
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    MOONRIVER 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.99 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 404.96 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    POLYGON 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 902.24 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    POLYGON_ZKEVM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 23.35 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    SCROLL 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 307.34 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    THUNDERCORE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 10.46 k
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)
    
    
    ZETA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 394
    End timestamp: 1720186136 (2024-07-05T13:28:56.000Z)

0xMasayoshi commented:
I added logic to use Sushi's pricing API for networks where defillama is missing data.

Filecoin, haqq, harmony, rootstock, and skale europa are throwing RPC erros when getting blocks.

Please take a look when you have a chance.

dtmkeng commented:
which name you need list  example `sushiswap aggregator` ?

0xMasayoshi commented:
> which name you need list example `sushiswap aggregator` ?

can we name it `SushiSwap`?

dtmkeng commented:
if need list name `SushiSwap` you should move to under `dexs/sushiswap` and create breakdown for agg
i prefer split name it easy for backfill

0xMasayoshi commented:
> if need list name `SushiSwap` you should move to under `dexs/sushiswap` and create breakdown for agg i prefer split name it easy for backfill

sure, moved to `dexs/sushiswap` and updated the BreakdownAdapter there

llamatester commented:
The sushiswap adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs sushiswap
    
    🦙 Running SUSHISWAP adapter 🦙
    ---------------------------------------------------
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    GRAPH_API_KEY env variable is not set, using the default api key
    Start Date:	Sat, 06 Jul 2024 15:53:01 GMT
    End Date:	Sun, 07 Jul 2024 15:53:01 GMT
    ---------------------------------------------------
    
    error fetching block [Error: Llama RPC error! methos: undefined 
    - host: https://zetachain-evm.blockpi.network/v1/rpc/public error: Cannot read properties of null (reading 'number')] {
      _underlyingError: '[object Object]',
      _isCustomError: true
    }
    GraphFetchError: Failed to get daily volume on celo with graph https://gateway-arbitrum.network.thegraph.com/api/5a1340b49fa9efe0021452daa260564e/subgraphs/id/8roCC7H2tsGYGvxD52QQbUoHXXx77H9tPhNn1qcjB5yj: subgraph not found: 8roCC7H2tsGYGvxD52QQbUoHXXx77H9tPhNn1qcjB5yj
    Attempting with alternative query...
    GraphFetchError: Failed to get daily volume on core with graph https://thegraph.coredao.org/subgraphs/name/sushi-v2/sushiswap-core: Type `Query` has no field `dayData`
    Attempting with alternative query...
    Failed to get alternative daily volume on celo with graph https://gateway-arbitrum.network.thegraph.com/api/5a1340b49fa9efe0021452daa260564e/subgraphs/id/8roCC7H2tsGYGvxD52QQbUoHXXx77H9tPhNn1qcjB5yj: subgraph not found: 8roCC7H2tsGYGvxD52QQbUoHXXx77H9tPhNn1qcjB5yj
    GraphFetchError: Failed to get daily volume on blast with graph https://api.goldsky.com/api/public/project_clslspm3c0knv01wvgfb2fqyq/subgraphs/sushiswap/sushiswap-blast/gn: Type `Query` has no field `dayData`
    Attempting with alternative query...
    Attempting with alternative query...
    GraphFetchError: Failed to get total volume on celo with graph https://gateway-arbitrum.network.thegraph.com/api/5a1340b49fa9efe0021452daa260564e/subgraphs/id/8roCC7H2tsGYGvxD52QQbUoHXXx77H9tPhNn1qcjB5yj: subgraph not found: 8roCC7H2tsGYGvxD52QQbUoHXXx77H9tPhNn1qcjB5yj
    Failed to get alternative daily volume on core with graph https://thegraph.coredao.org/subgraphs/name/sushi-v2/sushiswap-core: Type `Query` has no field `dayDatas`
    Failed to get alternative daily volume on blast with graph https://api.goldsky.com/api/public/project_clslspm3c0knv01wvgfb2fqyq/subgraphs/sushiswap/sushiswap-blast/gn: Type `Query` has no field `dayDatas`
    Unable to get daily fees on celo from graph.
    Unable to get total fees on celo from graph.
    Attempting with alternative query...
    Unable to get daily fees on blast from graph.
    Attempting with alternative query...
    Calculating volume excluding blacklisted tokens...
    Unable to get daily fees on core from graph.
    GraphFetchError: Failed to get total volume on polygon_zkevm with graph https://api.studio.thegraph.com/query/32073/v3-polygon-zkevm/v0.0.2: Failed to decode `block.number` value: `subgraph QmW1TpoQyP2LQCxgvY5JGt5fxr5dJFjLfTGKB3jkFUefiS has only indexed up to block number 80859 and data for block number 14023745 is therefore not yet available`
    error fetching block [Error: Llama RPC error! methos: undefined 
    - host: https://zetachain-evm.blockpi.network/v1/rpc/public error: Cannot read properties of null (reading 'number')] {
      _underlyingError: '[object Object]',
      _isCustomError: true
    }
    Calculating volume excluding blacklisted tokens...
    Version -> SUSHISWAP
    ---------
    ARBITRUM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 536.89 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    ARBITRUM_NOVA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 4.21 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    AVAX 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 15.57 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BASE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.24 M
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BLAST 👇
    Backfill start time: 1/3/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 70.14 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BOBA 👇
    Backfill start time: 22/3/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.98 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BOBA_BNB 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BSC 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 222.62 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BITTORRENT 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    CELO 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 36.58 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    CORE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 19.17 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    ETHEREUM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.37 M
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    FANTOM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.47 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    FUSE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.60 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    XDAI 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 130
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    KAVA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2.03 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    LINEA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 332.84 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    METIS 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 35
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    MOONBEAM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 32
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    MOONRIVER 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 6.05 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.08 M
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    POLYGON 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 588.14 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    POLYGON_ZKEVM 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 32.09 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    SCROLL 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 393.55 k
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    THUNDERCORE 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 2
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    ZETA 👇
    Backfill start time: 25/2/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 190
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    Version -> CLASSIC
    ---------
    ETHEREUM 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 20.26 M
    Total volume: 200.99 B
    Daily volume: 2.28 M
    Daily fees: 6.83 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 602.98 M
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 6.83 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 602.98 M
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 1.14 k
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 100.50 M
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 1.14 k
    └─ Methodology: None
    Total holders revenue: 100.50 M
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 5.69 k
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 502.48 M
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    BSC 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 40.27 M
    Total volume: 223.60 M
    Daily volume: 7.13 k
    Daily fees: 21
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 670.79 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 21
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 670.79 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 4
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 111.80 k
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 4
    └─ Methodology: None
    Total holders revenue: 111.80 k
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 18
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 558.99 k
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    POLYGON 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 59.07 M
    Total volume: 23.69 B
    Daily volume: 397.12 k
    Daily fees: 1.19 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 71.08 M
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 1.19 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 71.08 M
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 199
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 11.85 M
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 199
    └─ Methodology: None
    Total holders revenue: 11.85 M
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 993
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 59.23 M
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    ARBITRUM 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 229.75 M
    Total volume: 29.98 B
    Daily volume: 495.34 k
    Daily fees: 1.49 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 89.95 M
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 1.49 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 89.95 M
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 248
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 14.99 M
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 248
    └─ Methodology: None
    Total holders revenue: 14.99 M
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 1.24 k
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 74.96 M
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    CELO 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 26.54 M
    
    
    AVAX 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 47.67 M
    Total volume: 3.80 B
    Daily volume: 937
    Daily fees: 3
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 11.40 M
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 3
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 11.40 M
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 0
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 1.90 M
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 1.90 M
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 2
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 9.50 M
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    HARMONY 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 59.84 M
    Total volume: 2.96 B
    Daily volume: 5.38 k
    Daily fees: 16
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 8.88 M
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 16
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 8.88 M
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 3
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 1.48 M
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 3
    └─ Methodology: None
    Total holders revenue: 1.48 M
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 13
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 7.40 M
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    XDAI 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 34.85 M
    Total volume: 207.69 M
    Daily volume: 15.29 k
    Daily fees: 46
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 623.06 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 46
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 623.06 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 8
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 103.84 k
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 8
    └─ Methodology: None
    Total holders revenue: 103.84 k
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 38
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 519.22 k
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    BOBA 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 4.68 M
    Total volume: 11.52 M
    Daily volume: 7.88 k
    Daily fees: 24
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 34.55 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 24
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 34.55 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 4
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 5.76 k
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 4
    └─ Methodology: None
    Total holders revenue: 5.76 k
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 20
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 28.79 k
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    FUSE 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 30.45 M
    Total volume: 7.76 M
    Daily volume: 1.66 k
    Daily fees: 5
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total fees: 23.27 k
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Daily user fees: 5
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total user fees: 23.27 k
    └─ Methodology: Users pay a 0.3% fee on each trade
    Daily revenue: 1
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total revenue: 3.88 k
    └─ Methodology: A 0.05% of each trade goes to treasury
    Daily holders revenue: 1
    └─ Methodology: None
    Total holders revenue: 3.88 k
    └─ Methodology: None
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily supply side revenue: 4
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    Total supply side revenue: 19.39 k
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    CORE 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 15.69 M
    Total volume: 0
    Total fees: 0
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total user fees: 0
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total revenue: 0
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total holders revenue: 0
    └─ Methodology: None
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total supply side revenue: 0
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    BLAST 👇
    Backfill start time: 7/7/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 5.78 M
    Total volume: 0
    Total fees: 0
    └─ Methodology: SushiSwap charges a flat 0.3% fee
    Total user fees: 0
    └─ Methodology: Users pay a 0.3% fee on each trade
    Total revenue: 0
    └─ Methodology: A 0.05% of each trade goes to treasury
    Total holders revenue: 0
    └─ Methodology: None
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total supply side revenue: 0
    └─ Methodology: Liquidity providers get 5/6 of all trades in their pools
    
    
    FANTOM 👇
    Backfill start time not defined
    NO METHODOLOGY SPECIFIED
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 84.84 M
    Daily volume: 0
    Daily fees: 0
    Daily user fees: 0
    Daily protocol revenue: 0
    Daily supply side revenue: 0
    Daily holders revenue: 0
    Daily revenue: 0
    
    
    Version -> TRIDENT
    ---------
    POLYGON 👇
    Backfill start time: 7/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 25.94 M
    Total fees: 15.23 k
    Total user fees: 15.23 k
    Daily volume: 149
    Daily fees: 0
    Daily user fees: 0
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 7/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 46.82 M
    Total fees: 60.39 k
    Total user fees: 60.39 k
    Daily volume: 257
    Daily fees: 2
    Daily user fees: 2
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    ARBITRUM 👇
    Backfill start time: 7/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 47.41 M
    Total fees: 30.62 k
    Total user fees: 30.62 k
    Daily volume: 0
    Daily fees: 0
    Daily user fees: 0
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    BSC 👇
    Backfill start time: 7/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 5.14 M
    Total fees: 4.78 k
    Total user fees: 4.78 k
    Daily volume: 0
    Daily fees: 0
    Daily user fees: 0
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    AVAX 👇
    Backfill start time: 7/7/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 6.34 M
    Total fees: 4.11 k
    Total user fees: 4.11 k
    Daily volume: 0
    Daily fees: 0
    Daily user fees: 0
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    
    
    Version -> V3
    ---------
    ARBITRUM 👇
    Backfill start time: 1/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 229.75 M
    Total volume: 536.68 M
    Daily volume: 1.04 M
    Daily fees: 1.72 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 783.27 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 1.72 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 783.27 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 1.72 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 783.27 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    AVAX 👇
    Backfill start time: 4/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 47.67 M
    Total volume: 1.77 M
    Daily volume: 2.06 k
    Daily fees: 20
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 2.51 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 20
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 2.51 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 20
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 2.51 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    BSC 👇
    Backfill start time: 4/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 40.27 M
    Total volume: 28.56 M
    Daily volume: 6.54 k
    Daily fees: 4
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 19.61 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 4
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 19.61 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 4
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 19.61 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    BOBA 👇
    Backfill start time: 6/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 4.68 M
    Total volume: 1.50 M
    Daily volume: 0
    Total fees: 4.49 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total holders revenue: 0
    └─ Methodology: None
    Total user fees: 4.49 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total supply side revenue: 4.49 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    ETHEREUM 👇
    Backfill start time: 5/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 20.26 M
    Total volume: 192.68 M
    Daily volume: 162.12 k
    Daily fees: 54
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 225.28 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 54
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 225.28 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 54
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 225.28 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    FANTOM 👇
    Backfill start time: 4/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 84.84 M
    Total volume: 11.41 M
    Daily volume: 7.84 k
    Daily fees: 78
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 83.86 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 78
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 83.86 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 78
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 83.86 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    FUSE 👇
    Backfill start time: 4/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 30.45 M
    Total volume: 45.70 k
    Daily volume: 802
    Daily fees: 2
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 137
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 2
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 137
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 2
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 137
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    XDAI 👇
    Backfill start time: 5/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 34.85 M
    Total volume: 1.31 M
    Daily volume: 2.02 k
    Daily fees: 4
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 1.33 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 4
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 1.33 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 4
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 1.33 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    OPTIMISM 👇
    Backfill start time: 5/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 122.38 M
    Total volume: 43.09 M
    Daily volume: 12.99 k
    Daily fees: 35
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 82.81 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 35
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 82.81 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 35
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 82.81 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    POLYGON 👇
    Backfill start time: 4/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 59.07 M
    Total volume: 266.63 M
    Daily volume: 481.10 k
    Daily fees: 375
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 303.96 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 375
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 303.96 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 375
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 303.96 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    POLYGON_ZKEVM 👇
    Backfill start time: 6/4/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 14.02 M
    Daily volume: 0
    Total fees: 0
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    
    
    THUNDERCORE 👇
    Backfill start time: 17/5/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 168.25 M
    Total volume: 19.25 M
    Daily volume: 6.17 k
    Daily fees: 16
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 35.39 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 16
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 35.39 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 16
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 35.39 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    BASE 👇
    Backfill start time: 3/8/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 16.79 M
    Total volume: 679.66 M
    Daily volume: 782.62 k
    Daily fees: 482
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 2.52 M
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 482
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 2.52 M
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 482
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 2.52 M
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    CORE 👇
    Backfill start time: 21/7/2023
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 15.69 M
    Total volume: 645.55 M
    Daily volume: 17.68 k
    Daily fees: 13
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 303.52 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 13
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 303.52 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 13
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 303.52 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    
    
    BLAST 👇
    Backfill start time: 2/3/2024
    End timestamp: 1720367580 (2024-07-07T15:53:00.000Z)
    Block: 5.78 M
    Total volume: 1.90 M
    Daily volume: 4.19 k
    Daily fees: 1
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Total fees: 4.59 k
    └─ Methodology: Each pool charge between 0.01% to 1% fee
    Daily protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Total protocol revenue: 0
    └─ Methodology: Treasury receives a share of the fees
    Daily holders revenue: 0
    └─ Methodology: None
    Total holders revenue: 0
    └─ Methodology: None
    Daily user fees: 1
    └─ Methodology: Users pay between 0.01% to 1% fee
    Total user fees: 4.59 k
    └─ Methodology: Users pay between 0.01% to 1% fee
    Daily supply side revenue: 1
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Total supply side revenue: 4.59 k
    └─ Methodology: Liquidity providers get most of the fees of all trades in their pools
    Daily revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
    Total revenue: 0
    └─ Methodology: 0 to 1/4 of the fee goes to treasury
